src/Features/LanguageServer/Protocol/RoslynLanguageServer.cs

```diff
@@ -100,10 +104,33 @@ void AddBaseServiceFromFunc<T>(Func<ILspServices, object> creatorFunc)
             }
         }
 
-        public Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
+        public async Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
         {
             OnInitialized();
-            return Task.CompletedTask;
+
+            if (_capabilityRegistrationsProviders != null)
```

Comment:
Instead of `ICapabilityRegistrationsProvider` - I think I'd prefer having some kind of `IOnInitializedProvider` that gets triggered here.  There are likely other things someone might want to do on initialization, other than registering capabilities - being more generic would help that.

Reply:
this still applies